pages/en/lb4/Decorators.md

```diff
@@ -0,0 +1,316 @@
+---
+lang: en
+title: 'Decorators'
+keywords: LoopBack 4.0, LoopBack-Next
+tags:
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Decorators.html
+summary:
+---
+
+A decorator allows you to annotate or modify your class declarations and members with metadata.
+
+## Introduction
+
+*If you're new to Decorators in TypeScript, see [here](https://www.typescriptlang.org/docs/handbook/decorators.html) for more info.*
+
+Decorators give LoopBack the flexibility to modify your plain TypeScript classes 
+and properties in a way that allows the framework to better understand how to 
+make use of them, without the need to inherit base classes or add functions 
+that tie into an API.
+
+As a default, LoopBack comes with some pre-defined decorators:
+
+- [Route Decorators](#route-decorators)
+- [Dependency Injection](#dependency-injection)
+- [Authentication Decorator](#authentication-decorators)
+- [Repository Decorators](#repository-decorators)
+
+## Route Decorators
+
+Route decorators are used to expose controller methods as REST API operations.
+If you are not familiar with the concept Route or Controller, please see [LoopBack Route](routes.htm)
+and [LoopBack Controller](controllers.htm) to learn more about them.
+
+By calling a route decorator, you provide OpenAPI specification to describe the 
+endpoint which the decorated method maps to. You can choose different decorators 
+accordingly or do a composition of them:
+
+### API Decorator
+
+  Syntax: `@api`
```

Comment:
Should it be `@api(swaggerSpecObject)`?

Reply:
Maybe a link to the decorator apidocs?

Reply:
IIUC, these should show up in http://apidocs.loopback.io/@loopback%2fauthentication/ but I don't see them.  Do we require an npm publish to see them or is something else going on?

Reply:
@raymondfeng nice to know we have the apidocs already,will link the function there and add parameters. 
While as Rand says, seems most Route decorators are currently missing in http://apidocs.loopback.io/@loopback%2fcore/#1666